File: fdlsandbox-android-java/app/src/main/java/com/omatt/fdlsandbox/components/DeepLinkModule.java
package com.omatt.fdlsandbox.components;

import com.omatt.fdlsandbox.modules.DeepLinkPresenter;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by omarmatthew on 9/16/2017.
 * Module for DeepLinkActivity
 */
@Module
class DeepLinkModule {
    @Provides
    @Singleton
    DeepLinkPresenter provideDeepLinkPresenter(){
        return new DeepLinkPresenter();
    }
}

File: fdlsandbox-android-kotlin/app/src/main/java/com/omatt/fdlsandbox/components/DeepLinkModule.kt
package com.omatt.fdlsandbox.components

import com.omatt.fdlsandbox.modules.deeplink.DeepLinkPresenter
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by omarmatthew on 9/16/2017.
 * DeepLinkActivity Module
 */
@Module
class DeepLinkModule {
    @Provides
    @Singleton
    fun provideMainPresenter() = DeepLinkPresenter()
}